mdmix/core/models.py

```diff
@@ -0,0 +1,244 @@
+    replicas: List[ReplicaResponse]
+    total_replicas: int
+    successful_replicas: int
+    scripts_created: Dict[str, Path] = Field(default_factory=dict)
+    
+    def get_summary(self) -> Dict[str, Any]:
+        """Get project summary"""
+        return {
+            "project": self.project_name,
+            "status": self.status,
+            "total_replicas": self.total_replicas,
+            "successful": self.successful_replicas,
+            "failed": self.total_replicas - self.successful_replicas,
+            "directory": str(self.project_dir)
+        }
+
+
+class SolventInfo(BaseModel):
+    """Solvent information response"""
+    name: str
+    display_name: str
+    description: str = ""
+    density: float
+    molecular_weight: float
+    water_model: str = "TIP3P"
+    ionic: bool = False
+    residue_counts: Dict[str, int] = Field(default_factory=dict)
+    probes: Dict[str, Dict[str, Any]] = Field(default_factory=dict)
+
+
+class AnalysisRequest(BaseModel):
+    """Request model for analysis parameters"""
+    project_dir: Path
+    replicas: List[str] = Field(default_factory=list, description="Specific replicas to analyze")
+    probes: List[str] = Field(default_factory=list, description="Specific probes to analyze")
+    cutoff: float = Field(3.5, description="Probe cutoff distance")
+    grid_spacing: float = Field(0.5, description="Grid spacing for density calculation")
+    output_format: str = Field("dx", description="Output format for grids")
+
+
+# Template Variables Models
+class TemplateVariables(BaseModel):
+    """Base model for template variables"""
+    pass
+
+
+class MinimizationTemplateVars(TemplateVariables):
+    """Variables for minimization templates"""
+    maxcyc: int = 5000
+    ncyc: int = 2500
+    ntpr: int = 100
+    restraint_wt: float = 10.0
+    restraintmask: str = ""
+
+
+class EquilibrationTemplateVars(TemplateVariables):
+    """Variables for equilibration templates"""
+    nsteps: int = 25000
+    timestep: float = 2.0
+    temp0: float = 300.0
+    tempi: float = 100.0
+    ntpr: int = 1000
+    ntwx: int = 5000
+    restraint_wt: float = 5.0
+    restraintmask: str = ""
+    stage: int = 1
+
+
+class ProductionTemplateVars(TemplateVariables):
+    """Variables for production templates"""
+    nsteps: int = 500000
+    timestep: float = 2.0
+    temp0: float = 300.0
+    pres0: float = 1.0
+    ntpr: int = 1000
+    ntwx: int = 5000
+    ioutfm: int = 1  # NetCDF format
+    iwrap: int = 1  # Wrap coordinates
+    restraint_wt: float = 0.0
+    restraintmask: str = ""
+
+
+# Configuration Models (for YAML)
+class YAMLConfig(BaseModel):
+    """Base configuration model for YAML files"""
+    
+    class Config:
+        extra = "allow"
+        use_enum_values = True
+    
+    def to_yaml(self, path: Path):
+        """Save configuration to YAML file"""
+        import yaml
+        with open(path, 'w') as f:
+            yaml.dump(self.dict(), f, default_flow_style=False)
```

Comment:
i think the current way of doing that would be
```suggestion
            yaml.dump(self.model_dump(), f, default_flow_style=False)
```

Also, it is possible that we want to use pydantic-settings BaseSettings
https://docs.pydantic.dev/latest/concepts/pydantic_settings/